File: 0392-is-subsequence/0392-is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        int cs = 0;
        int ct = 0;
        while(ct < t.length() && cs < s.length()){
            if(s.charAt(cs) == t.charAt(ct)){
                cs++;
            }
            ct++;
        }
        return cs == s.length();
    }
}

File: 0392-is-subsequence/0392-is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        cs = 0
        ct = 0
        while cs < len(s) and ct < len(t):
            if s[cs] == t[ct]:
                cs += 1
            ct += 1
        return cs == len(s)